hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java

```diff
@@ -128,6 +149,26 @@ public void handle(@NonNull final HandleContext handleContext) {
         final var config = handleContext.configuration().getConfigData(ConsensusConfig.class);
         validateTransaction(txn, config, topic);
 
+        /* handle custom fees */
+        if (!topic.customFees().isEmpty() && !isFeeExempted(topic.feeExemptKeyList(), handleContext.keyVerifier())) {
+            // check payer limits or throw
+            if (!op.acceptAllCustomFees()) {
+                validateFeeLimits(topic.customFees(), op.maxCustomFees());
+            }
+            // create synthetic body and dispatch crypto transfer
+            final var syntheticBodies = customFeeAssessor.assessCustomFee(topic, handleContext);
+            for (final var syntheticBody : syntheticBodies) {
+                final var record = handleContext.dispatch(stepDispatch(
```

Comment:
As discussed at today's call, we need to add suppressing a separate record for a child dispatch.

Reply:
This is still under discussion and @JivkoKelchev created a seperate [PR](https://github.com/hashgraph/hedera-services/pull/17072) for that.